src/tests/multihost/sssd/testlib/common/utils.py

```diff
@@ -1684,28 +1677,60 @@ def remove_user_from_group(self, group, user):
+        add_ou = f"powershell.exe -inputformat none -noprofile '" \
+                 f"New-ADOrganizationalUnit  -Name \"automount\" " \
+                 f"-Path \"{self.ad_basedn}\" -Confirm:$False " \
+                 f"-ProtectedFromAccidentalDeletion $false'"
+        self.ad_host.run_command(add_ou, raiseonerr=False)
+
+        add_top = f"powershell.exe -inputformat none -noprofile '" \
+                  f"New-ADObject -Name \"auto.master\" -Path \"OU=automount," \
+                  f"{self.ad_basedn}\" -Type nisMap -OtherAttributes " \
+                  f"@{{\"nisMapName\"=\"auto.master\"}}'"
+        self.ad_host.run_command(add_top)
+
+        add_root = f"powershell.exe -inputformat none -noprofile '" \
+                   f"New-ADObject -Name \"/-\" -Path \"CN=auto.master," \
+                   f"OU=automount,{self.ad_basedn}\" -Type nisObject " \
+                   f"-OtherAttributes @{{\"nisMapName\"=\"auto.master\" ; " \
+                   f"\"nisMapEntry\"=\"auto.direct\"}}'"
+        self.ad_host.run_command(add_root)
+
+        add_direct = f"powershell.exe -inputformat none -noprofile '" \
+                     f"New-ADObject -Name \"auto.direct\" -Path \"" \
+                     f"OU=automount,{self.ad_basedn}\" -Type nisMap " \
+                     f"-OtherAttributes @{{\"nisMapName\"=\"auto.direct\"}}'"
+        self.ad_host.run_command(add_direct)
+
+        add_home = f"powershell.exe -inputformat none -noprofile '" \
+                   f"New-ADObject -Name \"auto.home\" -Path \"" \
+                   f"OU=automount,{self.ad_basedn}\" -Type nisMap " \
+                   f"-OtherAttributes @{{\"nisMapName\"=\"auto.home\"}}'"
+        self.ad_host.run_command(add_home)
+
     def add_map(self, name, nfs_server):
         """ Add a nisobject to auto.direct map """
-        # TODO: Rewrite in powershell if possible.
-        ad_conn_inst = self.ad_conn()
-        entrydn = 'cn=%s,cn=auto.direct,ou=automount,%s' % (name,
-                                                            self.ad_basedn)
-        nismapentry = '-fstype=nfs,rw %s:%s' % (nfs_server, name)
-        attr = {
-            'objectclass': [b'top', b'nisObject'],
-            'cn': name.encode('utf-8'),
-            'nisMapEntry': nismapentry.encode('utf-8'),
-            'nisMapName': b'auto.direct'}
-        (ret, _) = ad_conn_inst.add_entry(attr, entrydn)
+        entrydn = f'cn=auto.direct,ou=automount,{self.ad_basedn}'
+        nismapentry = f'-fstype=nfs,rw {nfs_server}:{name}'
+
+        add_map = f"powershell.exe -inputformat none -noprofile '" \
+                  f"New-ADObject -Name \"{name}\" -Path \"" \
+                  f"{entrydn}\" -Type nisObject " \
+                  f"-OtherAttributes @{{\"nisMapName\"=\"auto.direct\" ; " \
+                  f"\"nisMapEntry\"=\"{nismapentry}\"}}'"
+        add = self.ad_host.run_command(add_map, raiseonerr=False)
+        ret = 'Success' if add.returncode == 0 else 'Failure'
```

Comment:
Instead of just a 'Failure' message, can we add a more precise failure message? same for line 1733

Reply:
The returned values are based on the original (undescribed, undocumented) behavior. I would rather return empty string for failure to make the value Falsey or change the return type to bool and modify code using the function.
The exact failure mode should be visible in the log as the command is executed in powershell so returning the command output will not bring much of a value to the table.

Reply:
yeah but we can add like 'Failure while adding a nisobject' or 'Failure while removing nismap' something like this